src/utils/daos_hdlr.c

```diff
@@ -1566,6 +1567,1119 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+				path_length = snprintf(dpath, MAX_FILENAME, "%s", dst_filename);
+				if (path_length >= MAX_FILENAME) {
+					rc = -1;
+					fprintf(stderr, "Path length is too long.\n");
+					D_GOTO(out, rc);
+				}
+
+				mode_t tmp_mode_dir = S_IRWXU;
+				rc = file_mkdir_dfs(dst_file_dfs, dpath, &tmp_mode_dir);
+
+				/* Recursively call "fs_copy" with the new path. */
+				rc = fs_copy(src_file_dfs, dst_file_dfs, path, dfs_prefix_len, fs_dst_prefix);
+
+				/* set original source perms on directories after copying */
+				rc = file_chmod_dfs(src_file_dfs, path, st.st_mode);
+				if (rc != 0) {
+					fprintf(stderr, "updating source permissions failed on %s (%d)\n",
+						path, rc);
+					D_GOTO(out, rc);
+				}
+				rc = file_chmod_dfs(dst_file_dfs, dpath, st.st_mode);
+				if (rc != 0) {
+					fprintf(stderr, "updating destination permissions failed on %s (%d)\n",
+						dpath, rc);
+					D_GOTO(out, rc);
+				}
+
+			}
+		} else {
+			rc = ENOTSUP;
+			fprintf(stderr, "file type is not supported (%d)\n", rc);
+			D_GOTO(out, rc);
+		}
+	}
+
+	/* After going through all the entries, close the directory. */
+	if (file_closedir_dfs(src_file_dfs, src_dir))
+		fprintf(stderr, "Could not close '%s': %s\n",
+			dir_name, strerror (errno));
+out:
+	return rc;
+}
+
+static int
+fs_copy_connect(file_dfs_t *src_file_dfs,
+		   file_dfs_t *dst_file_dfs,
+		   struct cmd_args_s *ap,
+		   daos_cont_info_t *src_cont_info,
+		   daos_cont_info_t *dst_cont_info,
+		   struct duns_attr_t *src_dattr,
+		   struct duns_attr_t *dst_dattr)
+{
+	/* check source pool/conts */
+	int rc = 0;
+
+	/* open src pool, src cont, and mount dfs */
+	if (src_file_dfs->type == DAOS) {
+		rc = daos_pool_connect(ap->src_p_uuid, ap->sysname, NULL,
+				DAOS_PC_RW, &ap->pool, NULL, NULL);
+		if (rc != 0) {
+			fprintf(stderr, "failed to connect to destination pool: %d\n", rc);
+			D_GOTO(out, rc);
+		}
+		rc = daos_cont_open(ap->pool, ap->src_c_uuid, DAOS_COO_RW,
+			&ap->cont, src_cont_info, NULL);
+		if (rc != 0) {
+			fprintf(stderr, "failed to open source container: %d\n", rc);
+			rc = daos_pool_disconnect(ap->pool, NULL);
+			if (rc != 0) {
+				fprintf(stderr, "failed to disconnect from source pool "DF_UUIDF
+					": %s (%d)\n", DP_UUID(ap->src_p_uuid), d_errdesc(rc), rc);
+				D_GOTO(out, rc);
+			}
+		}
+		rc = dfs_mount(ap->pool, ap->cont, O_RDWR, &(src_file_dfs->dfs));
+		if (rc) {
+			fprintf(stderr, "dfs mount on source failed: %s, %d\n",
+				ap->src, rc);
+			rc = daos_pool_disconnect(ap->pool, NULL);
+			if (rc != 0) {
+				fprintf(stderr, "failed to disconnect from source pool "DF_UUIDF
+					": %s (%d)\n", DP_UUID(ap->src_p_uuid), d_errdesc(rc), rc);
+				D_GOTO(out, rc);
+			}
+			rc = daos_cont_close(ap->cont, NULL);
+			if (rc != 0) {
+				fprintf(stderr, "failed to close source container (%d)\n", rc);
+				D_GOTO(out, rc);
+			}
+		}
+	}
+
+	/* open dst pool, dst cont, and mount dfs */
+	if (dst_file_dfs->type == DAOS) {
```

Comment:
so if source open/connect failed, why do the destination?

Reply:
@mchaarawi if there is a failure connecting/opening src then it doesn't get to this point the function should return